src/cryptsetup/cryptsetup-generator.c

```diff
@@ -106,6 +106,38 @@ static int generate_keydev_mount(const char *name, const char *keydev, char **un
         return 0;
 }
 
+static int print_dependencies(FILE *f, const char* device_path) {
+        if (PATH_IN_SET(device_path, "/dev/urandom", "/dev/random", "/dev/hw_random"))
+                fputs("After=systemd-random-seed.service\n", f);
+        else if (!STR_IN_SET(device_path, "-", "none")) {
```

Comment:
please consider reversing this check, and then doing `return 0` to exit quickly. That way, the whole code below won't have to be indented...